Sources/Architecture/X86/HardwareBreakpointManager.cpp

```diff
@@ -197,6 +197,11 @@ int HardwareBreakpointManager::getAvailableLocation() {
   return (it - _locations.begin());
 }
 
+int HardwareBreakpointManager::findLocation(Address const &addr) {
```

Comment:
I'm not sure I understand the point of this function. Every time you use it, you check if the result is `>= 0` and then do stuff with the return index if it is. How is it different than calling `std::find()` and checking if the result is `!= _locations.end()`? You can then use the returned iterator and do `*it = myStuff;`.

Looks like not having this helper function does not make the code more complex, and avoids adding a function and putting it in the interface of the class.